tests/proxy_overload_recovery.erl

```diff
@@ -0,0 +1,442 @@
+                      "transient and print model states ===\n", []),
+            check(),
+            Res
+    end.
+
+%%
+%% Runners - on failure may be helpful to connect to the node and run
+%% check / recheck to verify a fix or try reshrinking.
+%%
+run() ->
+    run(60).
+
+run(Seconds) ->
+    eqc:quickcheck(eqc:testing_time(Seconds, ?MODULE:prop_proxy_recovery())).
+
+trap_run(Seconds) ->
+    eqc:quickcheck(eqc:testing_time(Seconds, ?TRAPEXIT(?MODULE:prop_proxy_recovery()))).
+
+check() ->
+    eqc:check(eqc_statem:show_states(?MODULE:prop_proxy_recovery())).
+
+recheck() ->
+    eqc:recheck(?MODULE:prop_proxy_recovery()).
+
+cover(Seconds) ->
+    cover:start(),
+    cover:compile(?MODULE),
+    eqc:quickcheck(eqc:testing_time(Seconds, ?MODULE:prop_proxy_recovery())),
+    cover:analyze_to_file(?MODULE),
+    cover:stop().
+
+%%
+%% Quickcheck Property - setup and run the statem
+%%
+prop_proxy_recovery() ->
+    ?FORALL(Cmds,commands(?MODULE),
+            begin
+                Start = os:timestamp(),
+                {H,S,Res} = run_commands(?MODULE, Cmds),
+                End = os:timestamp(),
+                Msecs = timer:now_diff(End, Start) div 1000,
+                ?WHENFAIL(
+                   begin
+                       RT = S#tstate.rt,
+                       case RT of
+                           undefined ->
+                               ok;
+                           #rt{ppid = PPid, vpid = VPid} ->
+                               eqc:format("\n\nPPid state = ~w\n",
+                                          [catch sys:get_status(PPid)]),
+                               eqc:format("Vnode MsgQLen = ~p\n",
+                                          [catch msgq_len(VPid)])
+                       end
+                   end,
+                   measure(duration, Msecs, 
+                   aggregate(with_title("Commands"), command_names(Cmds),
+                             pretty_commands(?MODULE, Cmds, {H, S, Res},
+                                             Res == ok))))
+            end).
+%%
+%% Generators
+%%
+pos_int() ->
+    ?LET(X, int(), 1 + abs(X)).
+
+%% Generate the threshold seed - see get_params
+threshold_seed() ->
+    frequency([%% test proxy disabled - 1%
+               {1, undefined},
+               %% test with all parameters given
+               {4, {choose(1,25),choose(1,25),choose(1,50)}},
+               %% test with the interval and threshold
+               {5, {choose(1, 50), choose(1, 50)}},
+               %% just provide a threshold, defaults for others
+               {90, choose(4, 100)}]).
+
+%%
+%% initial_state
+%%
+initial_state() ->
+    #tstate{}.
+
+%%
+%% Run prepare once before any other calls and then never again.
+%%
+precondition_common(#tstate{rt = undefined}, {call, _M, F, _A}) ->
+    F == prepare;
+precondition_common(_, {call, _M, F, _A}) ->
+    F /= prepare.
+
+%% %% Make sure we're still running what we think we're running - uncomment 
```

Comment:
Which lines do you think are dead? 

The precondition match on rt = undefined makes sure it is always the only task that can be selected first. The second clause makes sure that prepare is never selected again once the runtime parameters are created.